File: storio-common-annotations-processor/src/main/java/com/pushtorefresh/storio/common/annotations/processor/generate/Generator.java
package com.pushtorefresh.storio.common.annotations.processor.generate;

import com.pushtorefresh.storio.common.annotations.processor.introspection.StorIOTypeMeta;
import com.squareup.javapoet.JavaFile;

import org.jetbrains.annotations.NotNull;

public interface Generator<TypeMeta extends StorIOTypeMeta> {

    JavaFile generateJavaFile(@NotNull TypeMeta typeMeta);
}

File: storio-common-annotations-processor/src/main/kotlin/com/pushtorefresh/storio/common/annotations/processor/generate/Generator.kt
package com.pushtorefresh.storio.common.annotations.processor.generate

import com.pushtorefresh.storio.common.annotations.processor.introspection.StorIOTypeMeta
import com.squareup.javapoet.JavaFile

interface Generator<in TypeMeta : StorIOTypeMeta<*, *>> {

    fun generateJavaFile(typeMeta: TypeMeta): JavaFile
}